### ISSUE TITLE
Using __subclasses__ on Protocols produces typecheck error 

### ISSUE THREAD

**Issue by SandyChapman** [microsoft/pyright#4593]
**Describe the bug**
I'm seeing an unexpected result when I use the `__subclasses__` function on a protocol class. The result of executing `__subclasses__` on a Protocol base class is to return all types that explicitly subclass it. However, pyright complains that `"X" is not a concrete class type and cannot be assigned to type "Type[X]"`

**To Reproduce**

Typecheck the following code:

```python
from typing import List, Protocol, Type

class X(Protocol):
    x: str

class X1(X):
    x: str

class X2(X):
    x: str

xs: List[Type[X]] = X.__subclasses__()  # Error: "X" is not a concrete class type and cannot be assigned to type "Type[X]"
print(xs) # prints [<class '__main__.X1'>, <class '__main__.X2'>]
```

**Expected behavior**
I expect there to be no type check error here. I understand that the semantics of using subclasses on Protocols may lead developers to believe that implicit implementations of the protocol can be returned, but there's no reason to report this as an error IMO (it may be more suitable as a warning).

**Screenshots or Code**
The equivalent code not using a Protocol base works as expected. E.g.

```python
from typing import List, Protocol, Type

class Y:
    y: str

class Y1(Y):
    y: str

class Y2(Y):
    y: str

ys: List[Type[Y]] = Y.__subclasses__()  # no error here.
print(ys)
```

**VS Code extension or command-line**
This error is reported both in VS Code and using the pyright CLI.

![image](https://user-images.githubusercontent.com/3920249/217298093-54484353-8d4d-4ec9-8c8a-eeb61f323c54.png)

**Comment by erictraut** [comment]
PEP 544 (which introduced Protocols) is pretty explicit about the intended behavior here:

> Variables and parameters annotated with Type[Proto] accept only concrete (non-protocol) subtypes of Proto. The main reason for this is to allow instantiation of parameters with such type.

It provides examples that indicate that a type checker should treat this as an error. So I think pyright is doing the right thing here.

I can offer a couple of workarounds:
1. Remove the type declaration `list[type[X]]` from the variable `xs`.
2. Use a type annotation of `List[Callable[[], X]]`, which is not disallowed by PEP 544.
3. Use `cast` or `# type: ignore` to suppress the error.

**Comment by SandyChapman** [comment]
Thanks for the response @erictraut. I'm not surprised the behavior is outlined by a PEP.
